src/main/java/ca/corefacility/bioinformatics/irida/service/analysis/workspace/galaxy/SequenceFilePathType.java

```diff
@@ -0,0 +1,54 @@
+package ca.corefacility.bioinformatics.irida.service.analysis.workspace.galaxy;
+
+import java.nio.file.Path;
+import java.util.Objects;
+
+import ca.corefacility.bioinformatics.irida.model.irida.IridaSequenceFile;
+import ca.corefacility.bioinformatics.irida.model.workflow.execution.InputFileType;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Class linking together a sequence file path and the Galaxy file type.
+ *
+ */
+public class SequenceFilePathType {
+	private InputFileType fileType;
+	private Path path;
+
+	/**
+	 * Creates a new {@link SequenceFilePathType}.
+	 * 
+	 * @param sequenceFile The specific sequence file.
+	 */
+	public SequenceFilePathType(IridaSequenceFile sequenceFile) {
+		checkNotNull(sequenceFile, "sequenceFile is null");
+
+		this.path = sequenceFile.getFile();
+		this.fileType = sequenceFile.isGzipped() ? InputFileType.FASTQ_SANGER_GZ : InputFileType.FASTQ_SANGER;
```

Comment:
Think this is something we could just add to `SequenceFile` or `SequencingObject`?  It wouldn't need to be a property in the database, it could just be a method.  That way we can ditch the extra class too.

Reply:
The problem with adding to `SequenceFile` is that this is Galaxy-specific code (that is, `InputFileType` is specific to Galaxy). I did not want to mix Galaxy code with the IRIDA objects, so I kept the code separate.

Reply:
Understood, but this feels like an unnecessary wrapper class that could maybe be solved with just a static method call.  Previously I see you were passing around the `Path` instead of this `SequenceFilePathType`.  Couldn't you just continue sending around the `Path` and then check whether the file is zipped when you need it?  The only place I see the `InputFileType` is at https://github.com/phac-nml/irida/pull/399/files#diff-3f2e082b327e44d9660a84b42fb07165R202.  That could reduce this whole changeset down to just a method call.  Just replicate the functionality from `public boolean isGzipped()` in `SequenceFile`.

Thoughts on that?

Reply:
That makes sense. I've removed `SequenceFilePathType` in 6ed7026f0b624686040635d38893c171aab90801 and implemented what you mentioned.